sdk/php/src/ValueObject/Type.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dagger\ValueObject;
+
+use Dagger\Client\AbstractScalar;
+use Dagger\Client\IdAble;
+use Dagger\Exception\MissingSubtype;
+use Dagger\Exception\UnsupportedType;
+use Dagger\TypeDefKind;
+use DomainException;
+use ReflectionClass;
+use ReflectionNamedType;
+use ReflectionType;
+use RuntimeException;
+
+//@TODO support intersection types
+final readonly class Type
+{
+    public TypeDefKind $typeDefKind;
+
+    public function __construct(
+        public string $name,
+        public bool $nullable = false,
+    ) {
+        $this->typeDefKind = $this->getTypeDefKind($name);
+    }
+
+    public static function fromReflection(ReflectionType $type): self
+    {
+        if (!($type instanceof ReflectionNamedType)) {
+            throw new UnsupportedType(sprintf(
+                'Currently the PHP SDK only supports %s',
+                ReflectionNamedType::class,
+            ));
+        }
+
+        return new self($type->getName(), $type->allowsNull());
+    }
+
+    public function isIdable(): bool
+    {
+        if (!class_exists($this->name)) {
+            return false;
+        }
+
+        $class = new ReflectionClass($this->name);
+
+        return $class->implementsInterface(IdAble::class);
+    }
+
+    public function getShortName(): string
+    {
+        if (!class_exists($this->name)) {
+            throw new RuntimeException(sprintf(
+                'cannot get short class name from type: %s',
+                $this->name,
+            ));
+        }
+
+        $class = new ReflectionClass($this->name);
+
+        return $class->getShortName();
+    }
+
+    private function getTypeDefKind(string $nameOfType): TypeDefKind
+    {
+        switch ($nameOfType) {
+            case 'bool': return TypeDefKind::BOOLEAN_KIND;
+            case 'int': return TypeDefKind::INTEGER_KIND;
+            case 'string': return TypeDefKind::STRING_KIND;
+            case 'null':
+            case 'void': return TypeDefKind::VOID_KIND;
+        }
+
+        if ($nameOfType === 'array') {
+            throw new DomainException(sprintf(
+                '%s should not be constructed for arrays, use %s instead.' .
+                ' If this error occurred outside of developing the PHP SDK, it is a bug.',
+                self::class,
+                ListOfType::class,
+            ));
+        }
+
+        if (class_exists($nameOfType)) {
+            $parents = class_parents($nameOfType);
+            assert(is_array($parents));
+
+            if (in_array(AbstractScalar::class, $parents, true)) {
+                return TypeDefKind::SCALAR_KIND;
+            }
+
+            if (enum_exists($nameOfType)) {
+                return TypeDefKind::SCALAR_KIND;
+            }
```

Comment:
Enums have their own kind: `TypeDefKind::ENUM_KIND`.

Reply:
> Enums have their own kind: `TypeDefKind::ENUM_KIND`.

Ahah! Thank you, I hadn't rebased recently, so I didn't realise 0.12.0 added enum support. I'll get on that.

Reply:
Enums are now supported